### PR TITLE
[TT-2989] - Remote Runners using SDK

### PR DESCRIPTION
**Pull request by sainivedh**
## What

 - Contains [walkthrough](https://github.com/Clarifai/examples/blob/c3703f63cacf94aaf1ddcd4a031623b9cf275193/runners/inference_runner.ipynb) of how to serve inference runners models in local system using docker containers and python SDK
 - Added image_classification and text_genration types

### CODE REVIEW

File: runners/inference_runner.ipynb [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+   "metadata": {},
+   "source": [
+    "## Image Classification"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a Inference Runner"
```
Comment (zeiler) [inline runners/inference_runner.ipynb:64]: we need to add descriptions to each of these sections. It's a little weird to see how you have to make a runner and a remote-operator and they match based on labels within the same user/org account (that's important as we don't document that elsewhere).

File: runners/test-generation/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+RUN python3 -m pip install --no-cache-dir decord av==9.2.0
+
+# For `dinat` model
+RUN python3 -m pip install --no-cache-dir natten -f https://shi-labs.com/natten/wheels/$CUDA/
+
+# For `nougat` tokenizer
+RUN python3 -m pip install --no-cache-dir python-Levenshtein
+
+# For `clarifai`
+RUN python3 -m pip install --no-cache-dir clarifai
```
Comment (zeiler) [inline runners/test-generation/Dockerfile:74]: how come all these pip installs are separate lines in the dockerfile?

File: runners/test-generation/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,87 @@
+FROM nvidia/cuda:11.8.0-cudnn8-devel-ubuntu20.04
```
Comment (zeiler) [inline runners/test-generation/Dockerfile:1]: the folder of this file is called `test-generation` but I think you meant `text-generation`

**Review by zeiler — CHANGES_REQUESTED** [review]
Some comments to fix.

Going to slack you another runner that might be a good example too.

File: runners/test-generation/Dockerfile (hunk 3 above)
Comment (sainivedh) [reply runners/test-generation/Dockerfile]: my bad. Changed it to `text-generation`

File: runners/test-generation/Dockerfile (hunk 2 above)
Comment (sainivedh) [reply runners/test-generation/Dockerfile]: Followed the transformers GPU [Dockerfile](https://github.com/huggingface/transformers/blob/main/docker/transformers-all-latest-gpu/Dockerfile). Dependencies are grouped by functionality making it easier to follow

File: runners/inference_runner.ipynb (hunk 1 above)
Comment (sainivedh) [reply runners/inference_runner.ipynb]: Added descriptions where we create `runner` and `remote_operator` type model.

**Pull request closed without merging**